### PR TITLE
Updated infrastructure and tests

### PR DESCRIPTION
**Pull request by ipopescu**
While working on tests for session code, [following the docs](https://docs.casperlabs.io/dapp-dev-guide/writing-contracts/testing-session-code/), I realized this repo needs to be updated. Especially, the tests.

Changes:
- updated infrastructure of the repository
   - updated Cargo.toml files with new versions, new edition
   - updated rust-toolchain version
   - added .gitignore files
   - removed .idea files
   - removed unnecessary Cargo file in main folder
   - update Makefile 
- created a basic test file that compiles and runs successfully -- but this needs some meaningful testing of the associated account

### CODE REVIEW

File: contract/Cargo.toml [hunk 1]
```diff
@@ -2,21 +2,17 @@
 name = "contract"
 version = "0.1.0"
 authors = ["Karan Dhareshwar <karan@casperlabs.io>"]
```
Comment (ramujin) [inline contract/Cargo.toml]: You should add yourself as an author as well! :)

**Review by ramujin — APPROVED** [review]
Great work!

File: contract/Cargo.toml (hunk 1 above)
Comment (ipopescu) [reply contract/Cargo.toml]: :) maybe I spent too much time in enterprises, but I think a general authors list would be easier to maintain. I've made the update to this:
> authors = ["CasperLabs <https://discord.com/invite/Q38s3Vh>"]

**Comment by ipopescu** [comment]
Optimistically merging since I spoke with Karan and Zoltan about this over a call and slack.

**Pull request merged**